File: 1.0/src/cz/mg/vulkan/PFNvkGetImageMemoryRequirements2KHR.java
package cz.mg.vulkan;

public class PFNvkGetImageMemoryRequirements2KHR extends VkFunctionPointer {
    public PFNvkGetImageMemoryRequirements2KHR() {
    }

    protected PFNvkGetImageMemoryRequirements2KHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetImageMemoryRequirements2KHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetImageMemoryRequirements2KHR(long value) {
        setValue(value);
    }

    public PFNvkGetImageMemoryRequirements2KHR(VkInstance instance) {
        super(instance, new VkString("vkGetImageMemoryRequirements2KHR"));
    }

    public void call(VkDevice device, VkImageMemoryRequirementsInfo2 pInfo, VkMemoryRequirements2 pMemoryRequirements){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pInfo != null ? pInfo.getVkAddress() : VkPointer.NULL, pMemoryRequirements != null ? pMemoryRequirements.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, long pInfo, long pMemoryRequirements);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetImageMemoryRequirements2KHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetImageMemoryRequirements2KHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pInfo, jlong pMemoryRequirements){
    (void)env;
    (void)jc;
    PFN_vkGetImageMemoryRequirements2KHR f = (PFN_vkGetImageMemoryRequirements2KHR)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkImageMemoryRequirementsInfo2*)jniLongToPointer(pInfo)),
        ((VkMemoryRequirements2*)jniLongToPointer(pMemoryRequirements))
    );
}

